### ISSUE TITLE
oracle can't connect

### ISSUE THREAD

**Issue by mckennagene** [xo/usql#273]
just did a fresh brew install on Mac, trying to connect to Oracle using a url of this form ...
usql oracle://user:password:host:port/SID

and i just get this non-helpful reply
error: oracle: EOF

have installed about 3 other sql command lines and can get all of them to connect, but really want usql. any way to get more info about the specific error? i don't see a verbose mode

**Comment by kenshaw** [comment]
I'd suggest using the `godror` driver. The pure Go `go-ora` driver is fairly new and has issues like this.

**Comment by mckennagene** [comment]
Thank you! 

The godror works. I am still having some minor issues with the URL, I think.

when i connect via ...

./usql gr://user:password@host:port/sid

I get this error

Connected with driver godror (<unknown, error: godror: ORA-00942: table or view does not exist>)
Type "help" for help.

i can then run queries just fine, but the auto-complete doesn't work, when i start typing a table name and press  "tab" i get 

ERROR: 2021/07/16 13:37:01 Error getting selectables dpiStmt_execute: ORA-00942: table or view does not exist
ERROR: 2021/07/16 13:37:02 Error getting selectables dpiStmt_execute: ORA-00911: invalid character
ERROR: 2021/07/16 13:37:02 Error getting selectables dpiStmt_execute: ORA-00911: invalid character

**Comment by kenshaw** [comment]
What version of Oracle Database are you using? The autocomplete (ie, the metadata reader/writers) may not have the proper permissions for the user, and/or the version of your database might be incompatible with the current metadata implementation. Unfortunately, it's extremely difficult and challenging to try to make `usql` work with all versions of all databases. Realistically, we only target the current versions of major databases.

**Comment by kenshaw** [comment]
BTW -- if you're doing real/heavy development with Oracle with `usql`, please note that you can escape semicolons using the `\;`:

```sql
  CREATE PROCEDURE a_1_in_1_out(a_param IN INTEGER, a_return OUT INTEGER) AS    
  BEGIN    
    SELECT a_param INTO a_return FROM dual\;    
  END\;;
```
I do wish that the other database developers eventually support `$$` style strings, which is much more elegant than having to escape semi-colons everywhere.

**Comment by mckennagene** [comment]
Thank you for that. I have oracle version 12C, so it's getting on in years. I will check permissions. The autocomplete is more nice-to-have anyway. The features that brought me to usql over sqlplus and SQLcl is the support for standard bash keyboard shortcuts, ctrl-a,e,d,k, etc.

**Comment by nineinchnick** [comment]
There were two PRs merged recently that should fix autocomplete for Oracle: #272 and #276. Could you run `usql` from the master branch and possibly close this issue if it's working now?

**Comment by mckennagene** [comment]
Thanks!
